app/views/ErrorPage.scala.html

```diff
@@ -65,13 +67,20 @@ <h1 id="title" class="govuk-heading-xl">
     @if(code == 63088 || code == 64989 || code == 10003) {
         <p class="govuk-body">
             @Html(messages("ServiceMessage." + (code) + ".2"))
+            <script @{CSPNonce.attr}>document.querySelectorAll("#link-back-choose").forEach(elem => elem.addEventListener("click", () => window.history.go(-1)));</script>
         </p>
     }
 
     <div class="govuk-form-group">
         @code match {
-            case 63082 => {<a class="govuk-link" id="link-try-again" href='javascript: window.history.go(-1)'>@messages("Service.tryagain")</a>}
-            case 63090 | 63092  => {<a class="govuk-link" id="link-add-another" href='javascript: window.history.go(-1)'>@messages("Service.excludeanotheremployee")</a>}
+            case 63082 => {
+                <a class="govuk-link" id="link-try-again">@messages("Service.tryagain")</a>
+                <script @{CSPNonce.attr}>document.getElementById("link-try-again").addEventListener("click", () => window.history.go(-1));</script>
+            }
+            case 63090 | 63092  => {
+                <a class="govuk-link" id="link-add-another">@messages("Service.excludeanotheremployee")</a>
+                <script @{CSPNonce.attr}>document.getElementById("link-add-another").addEventListener("click", () => window.history.go(-1));</script>
```

Comment:
These 2 scripts are basically doing the same thing. Is there anyway of combining them? Also, would it not be best to keep all Script tags together towards the top of the body?

Reply:
true they do the same. However they're not going to be added at the same time. we might hit either of those errors and in such an instance we have a script to apply CSP on it. (These are some of the errors where error message has javascript code in it.) Anyway, if we add the tag without the anchor tag rendered on the page first, script tag will throw an error. so had to implement it like that